lib/mpl_toolkits/tests/test_axisartist_axislines.py

```diff
@@ -90,3 +90,18 @@ def test_ParasiteAxesAuxTrans():
         ax1.set_ylim((0, 5))
 
     ax2.contour(xx, yy, data, colors='k')
+
+
+@image_comparison(['axisline_style.png'], remove_text=True, style='mpl20')
+def test_axisline_style():
+    fig = plt.figure(figsize=(2, 2))
+    ax = fig.add_subplot(axes_class=AxesZero)
+    ax.axis["xzero"].set_axisline_style("-|>,size=3.0")
+    ax.axis["xzero"].set_visible(True)
+    ax.axis["yzero"].set_axisline_style("->")
+    ax.axis["yzero"].set_visible(True)
+
+    for direction in ("left", "right", "bottom", "top"):
+        ax.axis[direction].set_visible(False)
+
+    fig.tight_layout()
```

Comment:
Seems like this doesn't take into account the larger arrow head?

Reply:
Yeah, I also noted that. Not sure what the best way to go about it is? Maybe use a slightly smaller arrow head for now and open an issue for it? If the size is not changed, it will fit. The purpose was more to test changing the size rather than tight layout of it...

Reply:
I reduced the size and skipped the tight layout. Opening an issue about it.

Reply:
If possible, it would be nice to fix at the same time, so we can only add in the one image change.

Reply:
I looked into it and isn't there a fundamental problem here? The Axis should adjust and not have a fixed size? I tried adding the line to the tight_bbox method, but the results were not appealing (although "correct").

(There is also something strange with the ticks and labels which are not taken into account here.)

Btw, the reason I turned it into a draft is that I added the possibility to set the facecolor of the filled arrow. It is currently blue (the default color of however that is determined) and that is probably not the idea. Not sure if it should be like that or if one just should use the same color as the line? (Will need to improve PR title and possibly an API change note.)

Anyway, I think that there are two things here: one to test the axisline style (this PR), the other to test that the tight layout is working properly with axis artists in general.

Reply:
> I looked into it and isn't there a fundamental problem here? The Axis should adjust and not have a fixed size? I tried adding the line to the tight_bbox method, but the results were not appealing (although "correct").
> 

@oscargus Can you elaborate what you mean?

Reply:
If I understand things correctly, the layout methods will consider the size (bounding boxes) of "all" objects except for Axis and then add space for Axis (and ticks and labels?),  where the Axis adjust the size to the space given. As the arrow is part of an Axis, one cannot really add the arrow to the part of the layout that should be considered, as this gives weird results, but if one doesn't there will be no space for it.

But I am not fully sure that there isn't a way around this, just that simply adding the arrow in the bounding box calculation is not the way to go to get it nicely looking.

The current tests work as there is enough space for the arrow to fit, but increasing the arrow size will not increase the size provided.

Reply:
Thanks. It's been a while since I took a look at the `tight_layout` code but I though it compares bbox of the axes (the area that will be adjusted by subplot parameters) and the bbox that are occupied by 'all' of the artists including ticks and ticklabels, etc. And I think considering the arrow for the bbox calculation is not much different than considering ticks and ticklabels. And my initial fix seems to work okay. Is it similar to what you get and you think this is not appealing?

| w/o tight_layout| w/ tight_layout|
|----------|--------|
| ![axisline_style_size_color_no_tight](https://user-images.githubusercontent.com/95962/202163618-85d7f1b2-066a-4709-a723-7055a11c51f1.png)  | ![axisline_style_size_color_after_fix](https://user-images.githubusercontent.com/95962/202163507-f1ab22b9-2971-474c-96fd-b9285c9ec9e0.png) |

Reply:
That is much better than I got! Basically, I had a large distance between the last tick and the arrow, say 30% of the axis length.

Reply:
Here is my code. Feel free to pick it up.

https://github.com/leejjoon/matplotlib/tree/testaxislinestyle

I thought I can make PR to your feature branch but somehow I cannot.

Reply:
Thanks! I've added that commit to my branch. Not sure why you couldn't do it though...

It seems like it does not work when there are no grid labels, see the last test. Any idea?

Reply:
Sorry, never mind... (Good to call `tight_layout` to get a tight layout... Works like a charm!)